client/components/jetpack/connection-health/index.tsx

```diff
@@ -17,23 +23,35 @@ import {
 	XMLRPC_ERROR,
 } from './constants';
 import { ErrorNotice } from './error-notice';
-import { useCheckJetpackConnectionHealth } from './use-check-jetpack-connection-health';
 
 interface Props {
 	siteId: number;
 }
 
 export const JetpackConnectionHealthBanner = ( { siteId }: Props ) => {
 	const translate = useTranslate();
+	const dispatch = useDispatch();
 	const siteIsAutomatedTransfer = useSelector(
 		( state ) => !! isSiteAutomatedTransfer( state, siteId )
 	);
 
-	const {
-		data: jetpackConnectionHealth,
-		isLoading: isLoadingJetpackConnectionHealth,
-		isError: isErrorCheckJetpackConnectionHealth,
-	} = useCheckJetpackConnectionHealth( siteId );
+	const isErrorCheckJetpackConnectionHealth = useSelector( ( state ) =>
+		getJetpackConnectionHealthRequestError( state as AppState, siteId )
+	);
+
+	const isLoadingJetpackConnectionHealth = useSelector( ( state ) =>
+		isRequestingJetpackConnectionHealthStatus( state as AppState, siteId )
+	);
+
+	const jetpackConnectionHealth = useSelector( ( state ) =>
+		getJetpackConnectionHealth( state as AppState, siteId )
+	);
+
+	useEffect( () => {
+		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+		// @ts-ignore
+		dispatch( requestJetpackConnectionHealthStatus( siteId ) );
+	}, [ dispatch, siteId ] );
 
 	if (
 		isLoadingJetpackConnectionHealth ||
```

Comment:
My previous comment on this is still valid: there's no good reason to use `isLoadingJetpackConnectionHealth` at all, we can remove it. We can just rely on `jetpackConnectionHealth?.error` below - if it's nullish, then we don't need to do anything in this component anyway. That will allow us to remove:

* The `isLoadingJetpackConnectionHealth()` selector
* The `isLoading` reducer subtree
* The `JETPACK_CONNECTION_HEALTH_REQUEST_SUCCESS` action
* The related tests

Reply:
I see your point now that I am thinking about this better. Thanks for your feedback! I'll do the appropriate changes.